npm shrinkwrap on mac returns error codes

When testing on a mac, I found that the npm shrinkwrap was returning unexpected values and making the plugin not work.
I tracked it down to the line below
let { stdout, stderr} = await exec("npm shrinkwrap",{cwd:workspaceRoot}); within the runShrinkwrap method.
The command is returning empty string in stdout and is populating stderr on Mac OS.
e.g. "npm notice created a lockfile as npm-shrinkwrap.json. You should commit this file."
e.g. "npm notice package-lock.json has been renamed to npm-shrinkwrap.json. npm-shrinkwrap.json will be used for future installations."
These return a value of 0 from the call.
This is different to a true error which will return 1 and populate the error object.  
e.g. "npm ERR! Unexpected token : in JSON at position 14" if the package.json is ill formed etc.

--- a/src/packageJsonProcessor.ts
+++ b/src/packageJsonProcessor.ts
@@ -13,15 +13,17 @@ export class PackageJsonProcessor{
         }
         try{
             let { stdout, stderr} = await exec("npm shrinkwrap",{cwd:workspaceRoot});
-			if (stdout) {
+            let npmShrinkWrapFile = "npm-shrinkwrap.json";
+            let shrinkWrapSucceeded = stdout || stderr.search(npmShrinkWrapFile)>-1
+			if (shrinkWrapSucceeded) {
 				let lines = stdout.split(/\r{0,1}\n/);
                 //read npm-shrinkwrap.json
-				let obj = JSON.parse(fs.readFileSync(path.join(workspaceRoot,"npm-shrinkwrap.json"), "utf8"));
+				let obj = JSON.parse(fs.readFileSync(path.join(workspaceRoot, npmShrinkWrapFile), "utf8"));
 				return this.flattenAndUniqDependencies(obj);
 			}
 		}
 		catch(e){
-            return Promise.reject('npm shrinkwrap failed, try running it manually to see what went wrong');
+            return Promise.reject('npm shrinkwrap failed, try running it manually to see what went wrong:' + e.error.message);
 		}
     }
 

